cmd/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/config"
+	"github.com/skupperproject/skupper/pkg/non_kube/apis"
+	"github.com/skupperproject/skupper/pkg/non_kube/common"
+	"github.com/skupperproject/skupper/pkg/non_kube/compat"
+	"github.com/skupperproject/skupper/pkg/non_kube/systemd"
+	"github.com/skupperproject/skupper/pkg/version"
+)
+
+var (
+	platform = config.GetPlatform()
+)
+
+func main() {
+	// if -version used, report and exit
+	isVersion := flag.Bool("version", false, "Report the version of the Skupper bootstrap command")
+	flag.Parse()
+	if *isVersion {
+		fmt.Println(version.Version)
+		os.Exit(0)
+	}
+
+	//
+	// NOTE FOR CONTAINERS
+	// When running bootstrap process through a container
+	// the /input path must be mapped to a directory containing a site
+	// definition based on CR files.
+	// It also expects the /output path to be mapped to the
+	// Host's XDG_DATA_HOME/skupper or $HOME/.local/share/skupper (non-root)
+	// and /usr/local/share/skupper (root).
+	//
+	fmt.Printf("Skupper nonkube bootstrap (version: %s)\n", version.Version)
+
+	var inputPath string
+	var outputPath string
+	inputPath = flag.Arg(0)
+
+	if platform.IsKubernetes() {
+		// Bootstrap uses podman as the default platform
+		_ = os.Setenv(types.ENV_PLATFORM, "podman")
+	}
+	if apis.IsRunningInContainer() {
+		inputPath = "/input"
+		outputPath = "/output"
+		for _, directory := range []string{inputPath, outputPath} {
+			stat, err := os.Stat(directory)
+			if err != nil {
+				fmt.Printf("Failed to stat %s: %s\n", directory, err)
+				os.Exit(1)
+			}
+			if !stat.IsDir() {
+				fmt.Printf("%s is not a directory\n", directory)
+				os.Exit(1)
+			}
+		}
+	} else {
+		binary := "podman"
+		if platform == types.PlatformSystemd {
+			binary = "skrouterd"
+		} else if platform == types.PlatformDocker {
+			binary = "docker"
+		}
+		err := exec.Command("command", "-v", binary).Run()
```

Comment:
This seems to work on my fedora workstation, but in general os/exec + shell builtins is a recipe for pain. If it must be done this way, I'd suggest exec.Command("/usr/bin/sh", "-c", ...). I think there may actually be a stdlib function for this exact check, though.

Reply:
Yes, `exec.LookPath` is definitely a better choice. I will update it.

Reply:
Done. Thank you @c-kruse.